scripts/zkai_d64_external_adapter_surface_probe.py

```
@@ -0,0 +1,550 @@
+    if source.get("mutation_suite") != {
+        "mutations_checked": 14,
+        "mutations_rejected": 14,
+        "decision": "GO",
+    }:
+        raise D64ExternalAdapterProbeError("source mutation suite drift")
+
+    requirements = payload["exact_semantic_requirements"]
+    if payload["exact_semantic_requirements_commitment"] != blake2b_commitment(
+        requirements, "ptvm:zkai:d64-external-adapter-requirements:v1"
+    ):
+        raise D64ExternalAdapterProbeError("semantic requirements commitment drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)